contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -428,33 +464,46 @@ transform_from_ci_as_for_likenode(Node *node, OpExpr *op, like_ilike_info_t like
 		Node	   *constant_suffix;
 		Const	   *highest_sort_key;
 
+		/* Always create a CollateExpr on top to match with op->inputcollid */
+		new_leftop = create_collate_expr(linitial(op->args), op->inputcollid);
+		linitial(op->args) = (Node*) new_leftop;
```

Comment:
can't we do this one line as we are doing just below this?

Reply:
Yes, updated in latest rev